ql/src/java/org/apache/hadoop/hive/ql/optimizer/physical/Vectorizer.java

```diff
@@ -1264,10 +1266,25 @@ private Support[] getVectorizedInputFormatSupports(
       try {
         InputFormat inputFormat = FetchOperator.getInputFormatFromCache(inputFileFormatClass, hiveConf);
         if (inputFormat instanceof VectorizedInputFormatInterface) {
+          // Special handling for Iceberg tables with ORC format.
+          if (inputFormat.getClass().getName().contains("HiveIcebergInputFormat")) {
+            boolean onlyOrcFiles = true;
+            Hive hiveDb = Hive.get(hiveConf, false);
+            for (String tableName : currentOperator.getSchema().getTableNames()) {
+              boolean isORCOnly =
+                  "TRUE".equalsIgnoreCase(hiveDb.getTable(tableName).getParameters().get("orc.files.only"));
```

Comment:
nit: might this look better? (not sure, depends on taste)
```
Boolean.valueOf(hiveDb.getTable(tableName).getParameters().get("orc.files.only"));
```

Reply:
Done